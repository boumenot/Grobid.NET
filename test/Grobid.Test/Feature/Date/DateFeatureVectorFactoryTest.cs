using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using Grobid.NET;
using Grobid.NET.Feature;
using Grobid.NET.Feature.Date;
using Xunit;

namespace Grobid.Test.Feature.Date
{
    [UseReporter(typeof(DiffReporter))]
    public class DateFeatureVectorFactoryTest
    {
        [Fact]
        public void DateFeatureExtractionTest()
        {
            var blockFactory = new RawBlockStateFactory();
            var blocks = blockFactory.CreateWithoutPunctuation("17 Nov 2009");

            var featureVectorFactory = new DateFeatureVectorFactory(
                new FeatureExtractor(EmptyLexicon.Instance));

            var featureVectors = blocks
                .Select(featureVectorFactory.Create)
                .ToArray();

            Approvals.VerifyAll(featureVectors, x => String.Join(" ", DateFeatureFormatter.Instance.Format(x)));
        }

        [Fact]
        [Trait("Category", "EndToEnd")]
        public void DateFeatureExtractionWapitiModelTest()
        {
            var blockFactory = new RawBlockStateFactory();
            var blocks = blockFactory.CreateWithoutPunctuation("17 Nov 2009");

            var featureVectorFactory = new DateFeatureVectorFactory(
                new FeatureExtractor(EmptyLexicon.Instance));

            var featureVectors = blocks
                .Select(featureVectorFactory.Create)
                .ToArray();

            using (var model = global::Wapiti.Wapiti.Load(@"content\models\date\model.wapiti"))
            {
                var lines = featureVectors.Select(x => String.Join(" ", DateFeatureFormatter.Instance.Format(x))).ToArray();
                Approvals.Verify(model.Label(lines));
            }
        }
    }
}
