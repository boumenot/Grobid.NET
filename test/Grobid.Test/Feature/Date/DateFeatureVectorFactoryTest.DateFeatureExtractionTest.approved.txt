17 17 1 17 17 17 7 17 17 17 LINEIN NOCAPS ALLDIGIT 0 0 0 NOPUNCT
Nov nov N No Nov Nov v ov Nov Nov LINEIN INITCAP NODIGIT 0 0 1 NOPUNCT
2009 2009 2 20 200 2009 9 09 009 2009 LINEIN NOCAPS ALLDIGIT 0 1 0 NOPUNCT
